src/app/job-list/job-list.component.ts

```diff
@@ -100,9 +109,13 @@ export class JobListComponent implements OnInit {
   onJobTypeChange(type) {
     if(type.value === 'scheduled') 
     {
-      this.dataSource = this.scheduledJobListData;
+      this.dataSource.data = this.scheduledJobListData;
+      this.dataSource.sort = this.sort;
+      setTimeout(() => this.dataSource.paginator = this.paginator);
     } else {
-      this.dataSource = this.draftJobListData;
+      this.dataSource.data = this.draftJobListData;
```

Comment:
@utraksh If Sort and paginator is same for draft and scheduled, can be outside if / else.